include/oneapi/tbb/concurrent_queue.h

```diff
@@ -131,8 +134,52 @@ class concurrent_queue {
         queue_allocator_traits::destroy(my_allocator, my_queue_representation);
         r1::cache_aligned_deallocate(my_queue_representation);
     }
+  
+    concurrent_queue& operator=( const concurrent_queue& other ){
+        //TODO: implement support for std::allocator_traits::propogate_on_container_copy_assignment
+        if (my_queue_representation != other.my_queue_representation) {
+            clear();
+	    my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
+	}
+	return *this;
+    }
+
+    concurrent_queue& operator=( concurrent_queue&& other ){
+        //TODO: implement support for std::allocator_traits::propogate_on_container_move_assignment
+        if (my_allocator == other.my_allocator){
+	    my_queue_representation = other.my_queue_representation;
+	}
+	else{
+	    clear();
```

Comment:
I guess `clear()` should be done in the beginning of the method unconditionally - it is also required for equal allocators, because if `clear` would not be done before assigning the representations - previous memory would be lost.